app/controllers/concerns/api/badge_achievements_controller.rb

```diff
@@ -0,0 +1,41 @@
+module Api
+  module BadgeAchievementsController
+    extend ActiveSupport::Concern
+
+    def index
+      achievements = BadgeAchievement.order(created_at: :desc).page(params[:page]).per(50)
+      render json: achievements
+    end
+
+    def show
+      achievement = BadgeAchievement.find(params[:id])
+      render json: achievement
+    end
+
+    def create
+      achievement = BadgeAchievement.new(badge_achievement_params)
+
+      if achievement.save
+        render json: achievement, status: :created
+      else
+        render json: { errors: achievement.errors.full_messages }, status: :unprocessable_entity
+      end
+    end
+
+    def destroy
+      achievement = BadgeAchievement.find(params[:id])
+      achievement.destroy
+      head :no_content
+    end
+
+    private
+
+    def badge_achievement_params
```

Comment:
Sorry one change I didn't describe earlier

Let's make these additionally permitted in the write params:

- `rewarding_context_message_markdown`
- `include_default_description`

These are extra params not always needed, but should be available to admins using this API.